app/feature_flags.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/config"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+// setupFeatureFlags called on startup and when the cluster leader changes.
+// starts or stops the syncronization of feature flags from upstream managment.
+func (s *Server) setupFeatureFlags() {
+	license := s.License()
+	inCloud := license != nil && *license.Features.Cloud
+	splitKey := *s.Config().ServiceSettings.SplitKey
```

Comment:
`s.Config().ServiceSettings.SplitKey` could never be `nil` right? Just checking.

Reply:
Yup. Config should exist here.